rskj-core/src/main/java/co/rsk/net/light/MessageVisitor.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net.light;
+
+import co.rsk.net.eth.LightClientHandler;
+import co.rsk.net.light.message.*;
+import io.netty.channel.ChannelHandlerContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by Julian Len and Sebastian Sicardi on 21/04/20.
+ */
+public class MessageVisitor {
```

Comment:
Since there is another MessageVisitor, even if it is on a different namespace, I'm tempted to change the name, (MessageVisitor isn't on itself very meaningful either), maybe to something like LightClientMessageVisitor?